src/ORM/FieldType/DBVarchar.php

```diff
@@ -117,12 +123,14 @@ public function RTF()
 
     public function scaffoldFormField($title = null, $params = null)
     {
-        if (!$this->nullifyEmpty) {
-            // Allow the user to select if it's null instead of automatically assuming empty string is
-            return new NullableField(new TextField($this->name, $title));
-        } else {
-            // Automatically determine null (empty string)
-            return parent::scaffoldFormField($title);
+        // Set field with appropriate size
+        $field = TextField::create($this->name, $title);
+        $field->setMaxLength($this->getSize());
+
+        // Allow the user to select if it's null instead of automatically assuming empty string is
+        if (!$this->getNullifyEmpty()) {
+            $field = NullableField::create($field);
```

Comment:
this if condition could be at the top and return early

Reply:
It can't be at the top because it needs the pre-constructed $field nested inside it.